src/dreyfus/src/dreyfus_util.erl

```diff
@@ -59,6 +59,15 @@ use_ushards(#index_query_args{stable=true}) ->
 use_ushards(#index_query_args{}) ->
     false.
 
+
+get_ring_opts(#index_query_args{partition = nil}, _Shards) ->
+    [];
+get_ring_opts(#index_query_args{}, Shards) ->
+    Shards1 = lists:map(fun(#shard{} = S) ->
+        S#shard{ref = undefined}
```

Comment:
Since we want to match those shards exactly we set the worker references to `undefined`. 

In `fabric_db_partition` we get them directly from mem:shards(...) so they won't have references set. In case of dreyfus in one case we get the shards from the workers fabric_dict which have references so we clear them out in `get_ring_opts`.

Reply:
Re: `_all_docs` there a `fabric_streams` abstraction used by those types of requests and there `RingOpts` is already passed in

 https://github.com/apache/couchdb/blob/master/src/fabric/src/fabric_view_all_docs.erl#L26

 https://github.com/apache/couchdb/blob/master/src/fabric/src/fabric_streams.erl#L49